### PR TITLE
Highlight newly scanned bp passports in `EditPatientScreen`

### PR DESCRIPTION
**Pull request by Honey14**
https://app.shortcut.com/simpledotorg/story/6834/add-highlight-when-bp-passports-are-added

### CODE REVIEW

File: app/src/main/java/org/simple/clinic/editpatient/EditPatientViewRenderer.kt [hunk 1]
```diff
@@ -82,7 +82,7 @@ class EditPatientViewRenderer(private val ui: EditPatientUi) : ViewRenderer<Edit
     val newlyAddedBpPassports = model.ongoingEntry.bpPassports?.map { it.displayValue() }.orEmpty()
     val identifiers = model.bpPassports?.map { it.identifier.displayValue() }.orEmpty()
 
-    ui.displayBpPassports(identifiers + newlyAddedBpPassports)
+    ui.displayBpPassports(identifiers, newlyAddedBpPassports)
```
Comment (msasikanth) [inline app/src/main/java/org/simple/clinic/editpatient/EditPatientViewRenderer.kt:85]: This approach works fine, but we end up passing two lists and then checking the highlight status at the view level. 

Instead, maybe we can map the identifiers to a specific type and then render those? Something like this.

```kt
val newlyAddedBpPassports = model.ongoingEntry.bpPassports
    ?.map { BpPassportListItem(identifier = it.displayValue(), isHighlighted = true) }
    .orEmpty()

val identifiers = model.bpPassports
    ?.map {
      BpPassportListItem(identifier = it.identifier.displayValue(), isHighlighted = false)
    }
    .orEmpty()
```

So, now we don't have to use nested loops to identify the highlight status at view level and can simply use the boolean value. This way we can actually have the tests to verify the highlight status as well. WDYT?

File: app/src/main/java/org/simple/clinic/editpatient/EditPatientScreen.kt [hunk 2]
```diff
@@ -563,6 +563,15 @@ class EditPatientScreen : BaseScreen<
     }
   }
 
+  private fun highlightNewlyScannedBpPassports(bpPassportView: PatientEditBpPassportViewBinding) {
+    bpPassportView.bpPassportIdentifier.setBackgroundColor(resources.getColor(R.color.simple_yellow_100))
```
Comment (msasikanth) [inline app/src/main/java/org/simple/clinic/editpatient/EditPatientScreen.kt:567]: Can we use `context.resolveColor` instead, please? `resources.getColor` is deprecated.

**Review by msasikanth — CHANGES_REQUESTED** [review]
Left couple of comments, PTAL

**Comment by janhavisinghh** [comment]
@msasikanth added the requested changes as new commits

File: app/src/main/java/org/simple/clinic/editpatient/EditPatientScreen.kt [hunk 3]
```diff
@@ -557,9 +557,8 @@ class EditPatientScreen : BaseScreen<
 
   private fun dateOfBirthFocusChanges(): Observable<EditPatientEvent> = dateOfBirthEditText.focusChanges.map(::DateOfBirthFocusChanged)
 
-  override fun displayBpPassports(identifiers: List<BPPassportListItem>, newlyScannedIdentifiers: List<BPPassportListItem>) {
+  override fun displayBpPassports(listOfBpPassports: List<BPPassportListItem>) {
```
Comment (msasikanth) [inline app/src/main/java/org/simple/clinic/editpatient/EditPatientScreen.kt:560]: Personally I would have preferred to just name it `bpPassports` 😅

Comment (janhavisinghh) [reply app/src/main/java/org/simple/clinic/editpatient/EditPatientScreen.kt:561]: haha. oki. Changing it.

**Review by msasikanth — APPROVED** [review]
LGTM

**Pull request merged**